src/storage/ducklake_metadata_manager.cpp

```diff
@@ -2097,32 +2097,101 @@ ORDER BY snapshot_id
 	return snapshots;
 }
 
-vector<DuckLakeFileScheduledForCleanup> DuckLakeMetadataManager::GetFilesScheduledForCleanup(const string &filter) {
-	auto res = transaction.Query(R"(
+vector<DuckLakeFileForCleanup> DuckLakeMetadataManager::GetOldFilesForCleanup(const string &filter) {
+	auto query = R"(
 SELECT data_file_id, path, path_is_relative, schedule_start
 FROM {METADATA_CATALOG}.ducklake_files_scheduled_for_deletion
-)" + filter);
+)" + filter;
+	auto res = transaction.Query(query);
 	if (res->HasError()) {
 		res->GetErrorObject().Throw("Failed to get files scheduled for deletion from DuckLake: ");
 	}
 	auto context = transaction.context.lock();
-	vector<DuckLakeFileScheduledForCleanup> result;
+	vector<DuckLakeFileForCleanup> result;
 	for (auto &row : *res) {
-		DuckLakeFileScheduledForCleanup info;
+		DuckLakeFileForCleanup info;
 		info.id = DataFileIndex(row.GetValue<idx_t>(0));
 		DuckLakePath path;
 		path.path = row.GetValue<string>(1);
 		path.path_is_relative = row.GetValue<bool>(2);
 		info.path = FromRelativePath(path);
 		info.time = GetTimestampTZFromRow(*context, row, 3);
-
 		result.push_back(std::move(info));
 	}
 	return result;
 }
+vector<DuckLakeFileForCleanup> DuckLakeMetadataManager::GetOrphanFilesForCleanup(const string &filter,
+                                                                                 const string &data_path,
+                                                                                 const string &separator) {
+	auto query = R"(SELECT filename
+FROM read_blob('{DATA_PATH}**')
+WHERE filename NOT IN (
+SELECT REPLACE(
+           CASE
+               WHEN NOT file_relative THEN file_path
+               ELSE CASE
+                        WHEN NOT table_relative THEN table_path || file_path
+                        ELSE CASE
+                                 WHEN NOT schema_relative THEN schema_path || table_path || file_path
+                                 ELSE '{DATA_PATH}' || schema_path || table_path || file_path
+                             END
+                   END
+           END,
+           '/',
+           '{SEPARATOR}'
```

Comment:
Separator handling shouldn't happen in the SQL query. All paths are defined to have `/` as a separator in DuckLake. You can use the `LoadPath` method to get the correct path when reading the result.